runtime/compiler/runtime/MetaData.cpp

```diff
@@ -700,6 +703,8 @@ mapsAreIdentical(
        mapCursor->getMapSizeInBytes() == nextMapCursor->getMapSizeInBytes() &&
        mapCursor->getRegisterMap() == nextMapCursor->getRegisterMap() &&
        !memcmp(mapCursor->getMapBits(), nextMapCursor->getMapBits(), mapCursor->getMapSizeInBytes()) &&
+       (nonmergeableBCI.find(mapCursor) == nonmergeableBCI.end() ||
+        mapCursor->isByteCodeInfoIdenticalTo(nextMapCursor)) &&
```

Comment:
Doesn't adding the `isByteCodeInfoIdenticalTo` check make the `!comp->getOption(TR_FullSpeedDebug)` check redundant (and the comment as well)? Can we replace the FSD check with this `isByteCodeInfoIdenticalTo` check and pull it out of this `||` statement?